lib/Backends.hs

```diff
@@ -0,0 +1,36 @@
+module Backends
+  ( supportedBackends
+  , backendPackedCodec
+  ) where
+
+import qualified Data.Text           as T
+import qualified Toml
+
+import           Backend.Vault.Kv    (VaultKvBackend)
+import           Toml                (TomlCodec)
+import           Backend             (BackendPacked (..), Backend (..))
+import qualified Data.HashMap.Strict as HS
+import Validation (Validation(Failure))
+
+backendPackedCodec
+  :: (T.Text -> Either Toml.TomlBiMapError (Toml.TomlEnv BackendPacked))
+  -> TomlCodec BackendPacked
+backendPackedCodec backends = Toml.Codec input output
```

Comment:
this function is parameterized over `backends :: (T.Text -> Either Toml.TomlBiMapError (Toml.TomlEnv BackendPacked))`.

But that parameter seems to always be `supportedBackends`. So maybe `backendPackedCodec` should use the `supportedBackends` value directly, and not require that it be passed as a parameter.